### PR TITLE
Update language and locale enums and javadocs update

### PR DESCRIPTION
**Pull request by samvaity**
This PR 
- updates the Language and Locale to expandable enums.
   - Allows users to know the accepted values, better than a lookup list.
   - For cases, where only a subset of values are supported we let the user pass the value and have the service handle it. (consistent approach with FormContentType).
- Javadoc and tests update.
- Known service issues to be fixed in preview 3 - https://github.com/Azure/azure-sdk-for-java/issues/18967

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/RecognizeContentOptions.java [hunk 1]
```diff
@@ -87,16 +87,16 @@ public String getLanguage() {
...
     }
 
     /**
-     * Get the custom page numbers for multi-page documents(PDF/TIFF), input the number of the
-     * pages you want to get OCR result.
-     * <p>For a range of pages, use a hyphen. Separate each page or
-     * range with a comma, ex - ["1-3"].</p>
+     * Get the custom page numbers for multi-page documents(PDF/TIFF). Input the number of the
+     * pages you want to get the recognized result for.
+     * <p>For a range of pages, use a hyphen, ex - ["1-3"]. Separate each page or a page
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/RecognizeContentOptions.java:98]: nit
```suggestion
     * <p>For a range of pages, use a hyphen, ex - ["1-3"]. Separate each page or page
```

**Review by kristapratico — APPROVED** [review]

**Review by maririos — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormRecognizerLanguage.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+     * @param name a name to look for.
+     * @return the corresponding FormRecognizerLanguage.
+     */
+    @JsonCreator
+    public static FormRecognizerLanguage fromString(String name) {
+        return fromString(name, FormRecognizerLanguage.class);
+    }
+
+    /** @return known FormRecognizerLanguage values. */
+    public static Collection<FormRecognizerLanguage> values() {
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormRecognizerLanguage.java:53]: do we need to have this values() method?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeBusinessCardFromUrl.java [hunk 3]
```diff
@@ -170,10 +171,11 @@ public static void main(final String[] args) {
             if (addresses != null) {
                 if (FieldValueType.LIST == addresses.getValue().getValueType()) {
                     List<FormField> addressesItems = addresses.getValue().asList();
-                    addressesItems.stream().forEach(addressesItem -> {
+                    addressesItems.forEach(addressesItem -> {
                         if (FieldValueType.STRING == addressesItem.getValue().getValueType()) {
                             String address = addressesItem.getValue().asString();
-                            System.out.printf("Address: %s, confidence: %.2f%n", address, addressesItem.getConfidence());
+                            System.out
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeBusinessCardFromUrl.java:177]: is the breaking line here intended?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerClientTestBase.java [hunk 4]
```diff
@@ -331,7 +332,7 @@ private static void validateFieldValueTransforms(FieldValue expectedFieldValue,
                         DateTimeFormatter.ofPattern("HH:mm:ss")), actualFormField.getValue().asTime());
                     break;
                 case STRING:
-                    if (actualFormField.getName() != "ReceiptType") {
+                    if (!Objects.equals(actualFormField.getName(), "ReceiptType")) {
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormRecognizerClientTestBase.java:335]: if (!"ReceiptType".equals(actualFormField.getName())) {

**Review by mssfang — APPROVED** [review]

**Pull request merged**